src/aegis_ai_ml/src/classifier/secbert_fine_tuning.py

```diff
@@ -0,0 +1,510 @@
+                yticklabels=target_names,
+            )
+            plt.title("SecBERT Security Severity Classification - Confusion Matrix")
+            plt.ylabel("True Label")
+            plt.xlabel("Predicted Label")
+            plt.tight_layout()
+            plt.savefig("secbert_confusion_matrix.png", dpi=300, bbox_inches="tight")
+            plt.show()
+            print("Confusion matrix saved as 'secbert_confusion_matrix.png'")
+
+        return accuracy, report, cm
+
+    def predict_severity(self, text):
+        """Predict severity for new text"""
+        if self.model is None or self.tokenizer is None:
+            raise ValueError("Model not trained or loaded!")
+
+        self.model.eval()
+        with torch.no_grad():
+            inputs = self.tokenizer(
+                text, truncation=True, padding=True, max_length=512, return_tensors="pt"
+            ).to(self.device)
+
+            outputs = self.model(**inputs)
+            predictions = torch.nn.functional.softmax(outputs.logits, dim=-1)
+            predicted_class = torch.argmax(predictions, dim=-1).item()
+            confidence = torch.max(predictions).item()
+
+        severity = self.label_encoder.inverse_transform([predicted_class])[0]
+        return severity, confidence
+
+
+def main():
+    # Initialize classifier
+    classifier = SecBERTSeverityClassifier()
+
+    try:
+        df = classifier.load_and_preprocess_data_from_local(
+            os.getenv("AEGIS_ML_CVE_DATA_DIR")
+        )
+        classifier.prepare_model_and_tokenizer()
+        train_dataset, val_dataset, test_dataset = classifier.prepare_datasets(df)
+
+        classifier.train_model(train_dataset, val_dataset)
+
+        accuracy, report, cm = classifier.evaluate_model(test_dataset)
+
+        # Test sample predictions
+        print("SAMPLE PREDICTIONS:")
+        print("=" * 60)
+
+        sample_texts = [
+            "Buffer overflow vulnerability allows remote code execution with system privileges",
+            "Cross-site scripting vulnerability in web interface allows session hijacking",
+            "Information disclosure through verbose error messages in logs",
+            "Denial of service through resource exhaustion in HTTP parser",
+        ]
+
+        for i, text in enumerate(sample_texts, 1):
+            severity, confidence = classifier.predict_severity(normalize_text(text))
+            print(f"\n{i}. Text: {text}")
+            print(
+                f"   Predicted impact: {severity.upper()} (confidence: {confidence:.3f})"
+            )
+
+        # Final results
+        print(f"\n{'=' * 60}")
+        print("FINAL RESULTS")
+        print("=" * 60)
+        print(f"Model: {classifier.model_name}")
+        print(f"Test Accuracy: {accuracy:.4f} ({accuracy:.1%})")
+        print(f"Target Classes: {list(classifier.label_encoder.classes_)}")
+
+        print("\n Output files:")
+        print("  • ./secbert_model/ - Trained model and tokenizer")
```

Comment:
The model was actually created in `./etc/models/secbert_model/` when I tested this in my env.

Reply:
Fix proposed in #129.